index.js

```diff
@@ -356,6 +356,19 @@
     ($.Undefined  = type0('Undefined'))
   ];
 
+  // Enum :: [Any] -> Type
+  $.Enum = function(constants) {
+    return {
+      type: 'sanctuary-def/Enum',
+      test: function(x) {
+        return $.Array(Any).test(constants) && constants.indexOf(x) !== -1;
```

Comment:
What role does `$.Array(Any).test(constants)` serve? Verifying that `constants` is an Array? If so, I don't think this is appropriate. It's an error to apply `$.Enum` to a non-Array value. If we're to type check, we should do so up front. Since we don't do this elsewhere, though, I suggest not checking the type of `constants` at this point. If we decide to do so we should create a pull request to add type checking to all our type constructors.